src/plugins/paste-image/index.ts

```diff
@@ -48,11 +49,25 @@ export default class PasteImage extends PluginController {
           })
         );
       if (!imageFiles) return;
-      if (!this.calc.isAnyExpressionSelected) {
-        if (document.activeElement !== document.body) return;
-        // Avoid images being inserted at the top of the expressions list when there is no selected expression
-        const lastExprId = this.calc.getExpressions().at(-1)?.id;
-        if (lastExprId) this.setFocusLocation(lastExprId);
+      const geoFolderId = this.cc.getGeometryFolder()?.id;
+      const folderIdDenyList = [ID_METADATA_FOLDER].concat(geoFolderId ?? []);
+      const isInOrIsSpecialFolder = (model: ItemModel) =>
+        folderIdDenyList.includes(model.id) ||
+        (model.folderId && folderIdDenyList.includes(model.folderId));
```

Comment:
The right logic here seems to be instead
```suggestion
const isInOrIsSpecialFolder = (model: ItemModel) => model.isHiddenFromUI
```
This way, you can't paste into _any_ secret folders normally, and you can never paste into the geometry folder. Also, if authorFeatures are enabled, you can paste into all secret folders.

Note `model.isHiddenFromUI` is mostly `model.secret && !cc.shouldShowAuthorFeatures()` but with a few more quirks. In particular collapsed folders' children are marked as `model.isHiddenFromUI = true`. I think that's reasonable enough. If a collapsed folder is selected, we'd paste after the entire folder.

Reply:
Wow, this is a really elegant solution! Thank you for your suggestion.